src/services/account/account.service.impl.ts

```diff
@@ -0,0 +1,35 @@
+import Web3Modal from 'web3modal';
+import { AccountService } from './account.service';
+import { ServiceResponse } from '../core/service-response.';
+import { AccountStoreImpl, stores } from '../../store';
+import { Service } from '../core/service';
+import { Types } from '@safient/core';
+import { User } from 'models';
+
+export class AccountServiceImpl extends Service implements AccountService {
+  readonly web3Modal: Web3Modal;
+  private readonly accountStore: AccountStoreImpl;
+
+  constructor() {
+    super();
+    this.accountStore = stores?.accountStore;
+    this.web3Modal = new Web3Modal({
+      cacheProvider: true,
+      theme: 'light',
+    });
+  }
+
+  async login(): Promise<ServiceResponse<Types.User | undefined>> {
```

Comment:
Is undefined required in `ServiceResponse<Types.User | undefined>` ?

Reply:
I think we can eliminate this.